nevergrad/optimization/optimizerlib.py

```diff
@@ -256,6 +257,82 @@ def __init__(
 CMA = ParametrizedCMA().set_name("CMA", register=True)
 DiagonalCMA = ParametrizedCMA(diagonal=True).set_name("DiagonalCMA", register=True)
 
+class _FCMA(base.Optimizer):
+
+    def __init__(
+            self,
+            parametrization: IntOrParameter,
+            budget: Optional[int] = None,
+            num_workers: int = 1,
+            scale: float = 1.0,
+            popsize: int = 0
+    ) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        self._scale = scale
+        self._popsize = popsize if popsize > 0 else \
+                            max(self.num_workers, 4 + int(3 * np.log(self.dimension)))
+        self._es: Optional[cmaes.Cmaes] = None
+        # delay initialization to ease implementation of variants
+        self.listx: tp.List[ArrayLike] = []
+        self.listy: tp.List[float] = []
+        self.to_be_asked: tp.Deque[np.ndarray] = deque()
+
+    @property
+    def es(self) -> cmaes.Cmaes:
+        if self._es is None:
+            self._es = cmaes.Cmaes(x0=np.zeros(self.dimension, dtype=np.float),
+                                   input_sigma=self._scale,
+                                   popsize=self._popsize, randn=self._rng.randn)
+        return self._es
+
+    def _internal_ask(self) -> ArrayLike:
+        if not self.to_be_asked:
+            self.to_be_asked.extend(self.es.ask())
+        return self.to_be_asked.popleft()
+
+    def _internal_tell(self, x: ArrayLike, value: float) -> None:
+        self.listx += [x]
+        self.listy += [value]
+        if len(self.listx) >= self.es.popsize:
+            try:
+                self.es.tell(self.listy, self.listx)
+            except RuntimeError:
+                pass
+            else:
+                self.listx = []
+                self.listy = []
+
+    def _internal_provide_recommendation(self) -> ArrayLike:
+        if self._es is None:
+            raise RuntimeError("Either ask or tell method should have been called before")
+        if self.es.best_x is None:
+            return self.current_bests["pessimistic"].x
+        return self.es.best_x  # type: ignore
+
+
+class ParametrizedFCMA(base.ConfiguredOptimizer):
+    """FCMA-ES optimizer, wrapping external implementation: https://github.com/dietmarwo/fast-cma-es
+
+    Parameters
+    ----------
+    scale: float
+        scale of the search
+    popsize: int
+        population size, should be n * self.num_workers for int n >= 1.
+        default is max(self.num_workers, 4 + int(3 * np.log(self.dimension)))
+    """
+
+    # pylint: disable=unused-argument
+    def __init__(
+        self,
+        *,
+        scale: float = 0.3,
+        popsize: int = 0
+    ) -> None:
+        super().__init__(_FCMA, locals())
+
+FCMA = ParametrizedFCMA().set_name("FCMA", register=True)
+MiniFCMA = ParametrizedFCMA(scale=0.1, popsize=13).set_name("MiniFCMA", register=True)
```

Comment:
These `Mini` variants are mostly used in benchmarks, if this is a variant you want to use in benchmarks, please move it to optimization/experimentalvariants.py (this is new from this week) instead.
In the future, we want to keep only the very main variants of the algorithms in optimizerlib (so `FCMA` but not `MiniFCMA`).

Reply:
Ok. Will move the variant. In the long run experiments will show what to use in aggregated optimizers like NGO. If scale and popsize become part of parametrization one variant is probably sufficient both for CMA and FCMA

Reply:
popsize will not be in the parametrization, the parametrization only provides information on how the optimization space is structured, so I expect that popsize will always remain as a configuration to the optimizer

Reply:
Then we should also leave scaling as configuration to the optimizer. You wrote above
" hence already handled in the parametrization which normalizes the space."
which leaves scaling as an additional parameter not related to the optimization space.

Reply:
cf discussion above about sigma? anyway I'd rather have both popsize and sigma for the time being.